src/meetingsession/DefaultMeetingSession.ts

```diff
@@ -116,13 +129,47 @@ export default class DefaultMeetingSession implements MeetingSession, Destroyabl
     if (isDestroyable(this.audioVideoController)) {
       await this.audioVideoController.destroy();
     }
+    if (isDestroyable(this.eventReporter)) {
+      await this.eventReporter.destroy();
+    }
 
     this._logger = undefined;
     this._configuration = undefined;
     this._deviceController = undefined;
     this.audioVideoFacade = undefined;
     this.audioVideoController = undefined;
     this.contentShareController = undefined;
+    this._eventReporter = undefined;
+  }
+
+  private setupEventReporter(
+    configuration: MeetingSessionConfiguration,
+    logger: Logger,
+    eventReporter?: EventReporter
+  ): void {
+    if (eventReporter) {
+      this._eventReporter = eventReporter;
+    } else {
+      const {
+        urls: { eventsIngestionURL },
+      } = configuration;
+      if (eventsIngestionURL) {
+        const {
```

Comment:
What is the advantage of using this syntax instead of just assign it as const eventsIngestionURL = configuration.urls.eventsIngestionURL? Tbh, i found this very hard to read :)

Reply:
😅 I had it as you say, but got comments we can do this to avoid I think more lines so changed it to follow this.

Reply:
Sorry my bad, I can change it to one line :)

```
const eventsIngestionURL = configuration.urls.eventsIngestionURL?
```